src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -211,6 +211,65 @@ cdef class _ServicerContext:
         self._rpc_state.disable_next_compression = True
 
 
+cdef class _SyncServicerContext:
+    """Sync servicer context for sync handler compatibility."""
+
+    def __cinit__(self,
+                  _ServicerContext context):
+        self._context = context
+        self._callbacks = []
+        self._loop = context._loop
+
+    def read(self):
```

Comment:
This is supposed to emulate the existing sync stack, right? Why does it include the `read` and `write` methods?

Reply:
Removed.

(I thought it might be valuable.)

Reply:
While I think the read/write API is valuable in general, I don't think we want to encourage people to write *more* non-`asyncio` code.

Reply:
Agree. People who use `read`/`write` API in sync function ends up in kind of unsupported state.